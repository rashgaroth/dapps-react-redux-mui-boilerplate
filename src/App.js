import './styles/global/App.css';
import RenderRoutes from 'routes/routes';
import { StyledEngineProvider, CssBaseline, ThemeProvider } from '@mui/material';
import { useSelector } from 'react-redux';
// eslint-disable-next-line import/no-named-as-default
import theme from 'themes';

function App() {
  const globalState = useSelector(state => state.globalReducer);
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme({ customization: globalState })}>
        <CssBaseline />
        <RenderRoutes />
      </ThemeProvider>
    </StyledEngineProvider>
  );
}

export default App;
