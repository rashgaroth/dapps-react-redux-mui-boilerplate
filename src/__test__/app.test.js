import renderer from 'react-test-renderer';
import App from '../App';

it('Describe App.js Tests', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
