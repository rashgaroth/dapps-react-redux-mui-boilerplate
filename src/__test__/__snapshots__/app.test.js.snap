// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Describe App.js Tests 1`] = `
<div
  className="App"
>
  <p>
    Unit Testing!
  </p>
</div>
`;

exports[`Describe App.js Tests 2`] = `
<div
  className="App"
>
  <p>
    Unit Testing!
  </p>
</div>
`;

exports[`Describe App.js Tests 3`] = `
<div
  className="App"
>
  <p>
    Unit Testing!
  </p>
</div>
`;
